src/components/CustomTable/CustomTable.tsx

```diff
@@ -0,0 +1,134 @@
+import { CustomTableRow } from '../CustomTableRow/CustomTableRow';
+import { FetchError } from '../../errors/FetchError';
+import { MouseEvent } from 'react';
+import { SortingDirection } from '../../common/utils/SortingDirection';
+
+export function CustomTable<T extends Record<PropertyKey, unknown>>(props: CustomTableProps<T>) {
+  const {
+    toolbar,
+    columnConfig,
+    fetchUrl,
+    selectedRowsIds,
+    page,
+    rows,
+    rowsPerPage,
+    setColumnConfig,
+    setSelectedRowsIds,
+    setPage,
+    setRowsPerPage,
+    setRows,
+  } = props;
+
+  const handleClick = (event: MouseEvent<unknown>, row: T) => {
+    const selectedIndex = selectedRowsIds.indexOf(
+      selectedRowsIds.find((item) => {
+        return row.id === item;
+      }) as string
+    );
+    let newSelectedRowsIds: readonly string[];
+
+    if (selectedIndex === -1) {
+      newSelectedRowsIds = [...selectedRowsIds, String(row.id)];
+    } else {
+      const firstPart = selectedRowsIds.slice(0, selectedIndex);
+      const secondPart = selectedRowsIds.slice(selectedIndex + 1);
+      newSelectedRowsIds = [...firstPart, ...secondPart];
+    }
+
+    setSelectedRowsIds(newSelectedRowsIds);
+  };
+
+  const pixelHeightPerRow = 53;
+  const rowsPerPageArray = [5, 10, 25];
+
+  const handleChangePage = (_event: MouseEvent<HTMLButtonElement> | null, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  function createColumnConfig(data: T) {
+    const columnConfigArr: ColumnConfig<keyof T>[] = [];
+
+    Object.keys(data).forEach((key) => {
+      columnConfigArr.push({ id: key as keyof T, label: key.toUpperCase(), sortDirection: SortingDirection.ASC });
+    });
+
+    return columnConfigArr;
+  }
+  //if we are not on the first page, we need to calculate the number of empty rows to be displayed
+  //in most cases this is zero, on the last page it is usually less than the number of rows per page
+  const emptyRows = page > 0 ? Math.max(0, (1 + page) * rowsPerPage - rows.length) : 0;
+
+  useEffect(() => {
+    fetch(fetchUrl)
+      .then((res) => {
+        return res.json();
+      })
+      .then((data: T[]) => {
+        if (data) {
+          const colConfigArr = createColumnConfig(data[0]);
+          setColumnConfig(colConfigArr);
+        }
+        return setRows(data);
+      })
+      .catch((error) => {
+        const errorMessage = JSON.stringify(error);
+        throw new FetchError(`Failed to fetch with error: ${errorMessage}`);
+      });
+  });
+
+  return (
+    <Box
+      width='100%'
+      height='100%'
+    >
+      {toolbar}
+
+      <TableContainer>
+        <Table>
+          {<CustomTableHead {...props} />}
+          <TableBody>
+            {rows.map((row, index) => {
+              return (
+                <CustomTableRow<T>
+                  key={row.id as string}
```

Comment:
I think `row.id` should be a `string`

Reply:
fixed in [44e72af](https://github.com/dorm-social-room-manager/dsrm-frontend/pull/38/commits/44e72afe781fbe2467eebdb5c74be32d6133e4b1)